modules/extensions/src/terrain/shader-module.ts

```diff
@@ -0,0 +1,143 @@
+import type {_ShaderModule as ShaderModule} from '@deck.gl/core';
+
+import type {Texture2D} from '@luma.gl/core';
+
+export type TerrainModuleSettings = {
```

Comment:
TSDoc on an exported type feels like a must, TSDoc each field is really helpful when using this type in other places.